lib/auth/apikey.ex

```diff
@@ -9,57 +12,79 @@ defmodule Auth.Apikey do
   schema "apikeys" do
     field :client_secret, :binary
     field :client_id, :binary
-    field :description, :string
-    field :name, :string
-    field :url, :binary
     field :person_id, :id
     field :status, :id
+    belongs_to :app, Auth.App
 
     timestamps()
   end
 
-  @doc false
-  def changeset(apikey, attrs) do
-    apikey
-    |> cast(attrs, [:client_id, :client_secret, :name, :description, :url, :person_id])
-    |> validate_required([:client_secret])
+  @doc """
+  `encrypt_encode/1` does exactly what it's name suggests,
+  AES Encrypts a string of plaintext and then Base58 encodes it.
+  We encode it using Base58 so it's human-friendly (readable).
+  """
+  def encrypt_encode(plaintext) do
+    plaintext |> Fields.AES.encrypt() |> Base58.encode()
   end
 
-  def change_apikey(%Apikey{} = apikey) do
-    Apikey.changeset(apikey, %{})
+  @doc """
+  `create_api_key/1` uses the `encrypt_encode/1` to create an API Key
+  that is just two strings joined with a forwardslash ("/").
+  This allows us to use a *single* environment variable.
+  """
+  def create_api_key(id) do
+    encrypt_encode(id) <> "/" <> encrypt_encode(id)
```

Comment:
Does this just duplicate the string? - am i missing something?

Reply:
Close. On first inspection this appears to be the case, but actually these two strings are _always_ different.
Because the **initialisation vector** changes each time `Aes.encrypt` is called, the `client_id` and `client_secret` will always be different. We _sign_ the JWT with the `client_secret` and any app using `auth_plug` sends their `client_id` when making requests. i.e. the `client_secret` is _never_ transmitted. It's basic pre-shared key security.  💭